### PR TITLE
feat(exec_command): add optional timeout_secs parameter with child kill on expiry

### LINKED ISSUE #1152: feat(exec_command): add optional timeout_secs parameter with child kill on expiry
## Problem

`exec_command` has no process-level execution deadline. If a child process hangs for any reason (corrupted heredoc, stalled network mount, blocked read), aptu-coder waits indefinitely. The MCP client eventually fires its own transport timeout (300s--3600s), but by then the child process is still running as an orphan and the user waited the full duration with no actionable error.

The `timed_out` field already exists on both `ShellOutput` and `MetricEvent` but is always `false` -- the code path that sets it does not exist yet.

## Root cause

`run_with_timeout` (lib.rs L3934) awaits `drain_task` with no deadline:

```rust
let drain_task = tokio::spawn(async move { /* reads stdout+stderr pipes */ });
drain_task.await.ok();  // no timeout -- blocks until child exits and pipes close
```

The 500ms timeout that exists is only for the post-drain `child.wait()`, which never fires if drain never finishes.

## Proposed fix

Add `timeout_secs: Option<u64>` to `ExecCommandParams`. Thread it through `run_exec_impl` to `run_with_timeout`. When `Some(secs)` and `secs > 0`, wrap the entire `run_with_timeout` body (drain + child.wait) with `tokio::time::timeout`. On `Elapsed`:

1. Call `child.start_kill().ok()` -- child is in scope inside `run_with_timeout`
2. Await `child.wait()` to reap the zombie
3. Call `drain_task.abort()` -- **mandatory**: `tokio::spawn` detaches on `JoinHandle` drop; dropping the outer future does NOT cancel the spawned task; `abort()` is required
4. Return partial output with `timed_out: true` in `ShellOutput`

In the `exec_command` handler: set `timed_out: output.timed_out` in the `MetricEvent` emit (today hardcoded `false`). Use `INTERNAL_ERROR` for the error response (`RequestTimeout` does not exist in rmcp 1.7.0 `ErrorCode`).

Default is `None` (no timeout), preserving current behaviour for all existing callers.

## Properties

- Backward compatible: `timeout_secs` absent or `null` deserialises as `None`; no behaviour change for existing callers
- No schema snapshots to update (none exist in tests/)
- Two-timeout interaction: when aptu-coder returns `CallToolResult::error` before the goose transport timeout fires, goose receives a normal tool response -- no MCP transport state issue
- MCP 2025-11-25 aligned: timeout error uses `CallToolResult::error` with `is_error: true` and `.with_meta(Some(no_cache_meta()))`
- Activates the existing `timed_out` field in JSONL metrics (always `false` today)

## Acceptance criteria

- `exec_command` with `timeout_secs: 1` and `command: "sleep 60"` returns `is_error: true` within ~1s
- `exec_command` with `timeout_secs: 10` and a fast command completes normally with `timed_out: false`
- `timeout_secs: 0` treated as `None` (no timeout)
- Child process is reaped (no zombie) after timeout
- `timed_out: true` recorded in JSONL MetricEvent on timeout
- Tests wrapped in `tokio::time::timeout` at the test level to guard against harness hang if the fix regresses

## Relation to Issue 1

Issue 1 (heredoc validation) is the targeted fix for the specific PR#677 failure mode. This issue is defence-in-depth: a general safety net for any command that hangs, regardless of cause. Both are independent and non-breaking.

### PR DESCRIPTION
**Pull request by clouatre**
## Summary
Add optional `timeout_secs` parameter to `exec_command` tool for setting maximum command execution time. When the timeout is exceeded, the child process is killed and the response indicates `timed_out: true`.

Closes #1152

## Changes
- `crates/aptu-coder/src/lib.rs`: Add `timeout_secs: Option<i64>` parameter to `ExecCommandParams`; add `timed_out: bool` field to `ShellOutput`; implement timeout logic in `run_with_timeout` using `tokio::time::timeout` wrapper; update handler to return error on timeout
- `crates/aptu-coder/tests/exec_command.rs`: Add integration tests for timeout firing, fast command completion, zero timeout handling, and backward compatibility

## Deviation Note
Changed `timeout_secs` type from `Option<u64>` to `Option<i64>` to avoid non-standard JSON Schema uint64 format. Negative values are treated as `None` (no timeout).

## Test Plan
- [x] Tests pass (38 exec_command integration tests; 262 total tests pass)
- [x] Linter clean (cargo clippy -- -D warnings)
- [x] Security scan clean (aptu scan-security: 0 issues)

### CODE REVIEW

File: crates/aptu-coder/src/lib.rs [hunk 1]
```diff
@@ -4040,25 +4098,68 @@ async fn run_with_timeout(
...
-        match tokio::time::timeout(std::time::Duration::from_millis(500), child.wait()).await {
-            Ok(Ok(s)) => (Some(s), false),
-            Ok(Err(_)) => (None, false),
-            Err(_) => {
-                child.start_kill().ok();
-                let _ = child.wait().await;
-                (None, true)
-            }
+    // Common post-exit drain handling (500ms grace for background processes)
+    let post_exit = async {
```
[the remark this answers is not in the document]
Comment (clouatre) [reply crates/aptu-coder/src/lib.rs:4104]: The 500ms grace period is pre-existing and unrelated to this PR. Out of scope here.

File: crates/aptu-coder/src/lib.rs [hunk 2]
```diff
@@ -4040,25 +4098,68 @@ async fn run_with_timeout(
...
+            match tokio::time::timeout(std::time::Duration::from_secs(secs as u64), exec_future)
+                .await
+            {
+                Ok(result) => result,
+                Err(_elapsed) => {
+                    // Kill the child process. Ignore errors: the process may have already exited.
+                    child.start_kill().ok();
+                    // NOTE: abort is a no-op if the task already completed; tokio guarantees
+                    //       that calling abort on a completed JoinHandle is safe.
+                    drain_abort.abort();
```
[the remark this answers is not in the document]
Comment (clouatre) [reply crates/aptu-coder/src/lib.rs:4150]: Fixed in 35b197d: replaced `secs as u64` with `u64::try_from(secs).unwrap_or(u64::MAX)`. No overflow ambiguity remains.
